XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -184,6 +184,22 @@ private Problem GetProblem(IXenConnection connection, XenRef<VM> vmRef, string[]
 
                         return new NoPVDrivers(this, vm);
 
+                    case Failure.VM_LACKS_FEATURE:
+                        vm = connection.Resolve<VM>(vmRef);

```

Comment:
These lines that resolve the vm and throw exception if null are repeated through each case. Can we move them before the switch block, e.g.
```
VM vm = connection.Resolve<VM>(vmRef);
if (vm == null)
    throw new NullReferenceException(exception[0]);
```